lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,26 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $source Path (`container/object') of other object to symlink this object to
+     * @return \Guzzle\Http\Message\Response
+     */
+    public function symlink($source)
```

Comment:
Hmm... if I'm understanding the implementation of this method correctly, it would modify an existing object (say, `a`) in the Object Store to add a X-Object-Manifest header to it, whose value would be another object (say, `b`). This seems dangerous as the existing object (`a`) could already have data in it, that is, not be a manifest object. In other words, there is nothing preventing a user from retrieving such an existing data object and calling `symlink` on it.

For this reason, I would recommend changing the semantics of this method to go the other way. Specifically this method should:
* take as argument the name of the link object (which would not yet exist),
* create the link object with a header `X-Object-Manifest` whose value would be `$this` object's path, and 
* return the newly-created link object.

To clarify the intent of this method to the user, you might want to name it `createSymlinkFrom` or similar.